src/python/pants/core/goals/check.py

```diff
@@ -46,10 +48,13 @@ def from_fallible_process_result(
         *,
         partition_description: str | None = None,
         strip_chroot_path: bool = False,
+        strip_formatting: bool = False,
         report: Digest = EMPTY_DIGEST,
     ) -> CheckResult:
         def prep_output(s: bytes) -> str:
-            return strip_v2_chroot_path(s) if strip_chroot_path else s.decode()
+            chroot = strip_v2_chroot_path(s) if strip_chroot_path else s.decode()
+            formatting = cast(str, colors.strip_color(chroot)) if strip_formatting else chroot
```

Comment:
I'm pretty sure this cast is safe, because [`strip_colors`](https://github.com/jonathaneunice/colors/blob/c965f5b9103c5bd32a1572adb8024ebe83278fb0/colors/colors.py#L122-L131) uses a `str` regex rather than a binary one?